satellite/gracefulexit/endpoint_test.go

```diff
@@ -464,6 +464,102 @@ func TestSuccessPointerUpdate(t *testing.T) {
+
+		var recNodeID storj.NodeID
+
+		response, err := processClient.Recv()
+		require.NoError(t, err)
+
+		switch m := response.GetMessage().(type) {
+		case *pb.SatelliteMessage_TransferPiece:
+			require.NotNil(t, m)
+
+			pieceReader, err := exitingNode.Storage2.Store.Reader(ctx, satellite.ID(), m.TransferPiece.OriginalPieceId)
+			require.NoError(t, err)
+
+			header, err := pieceReader.GetPieceHeader()
+			require.NoError(t, err)
+
+			orderLimit := header.OrderLimit
+			originalPieceHash := &pb.PieceHash{
+				PieceId:   orderLimit.PieceId,
+				Hash:      header.GetHash(),
+				PieceSize: pieceReader.Size(),
+				Timestamp: header.GetCreationTime(),
+				Signature: header.GetSignature(),
+			}
+
+			newPieceHash := &pb.PieceHash{
+				PieceId:   m.TransferPiece.AddressedOrderLimit.Limit.PieceId,
+				Hash:      originalPieceHash.Hash,
+				PieceSize: originalPieceHash.PieceSize,
+				Timestamp: time.Now(),
+			}
+
+			receivingIdentity := nodeFullIDs[m.TransferPiece.AddressedOrderLimit.Limit.StorageNodeId]
+			require.NotNil(t, receivingIdentity)
+
+			// get the receiving node piece count before processing
+			recNodeID = receivingIdentity.ID
+
+			// update pointer to include the new receiving node before responding to satellite
+			keys, err := satellite.Metainfo.Database.List(ctx, nil, 1)
+			require.NoError(t, err)
+			path := string(keys[0])
+			pointer, err := satellite.Metainfo.Service.Get(ctx, path)
+			require.NoError(t, err)
+			require.NotNil(t, pointer.GetRemote())
+			require.True(t, len(pointer.GetRemote().GetRemotePieces()) > 0)
+
+			pieceToRemove := make([]*pb.RemotePiece, 1)
+			pieceToAdd := make([]*pb.RemotePiece, 1)
+			pieces := pointer.GetRemote().GetRemotePieces()
+
+			for _, piece := range pieces {
+				if pieceToRemove[0] == nil {
+					pieceToRemove[0] = piece
+					continue
+				}
+			}
+
+			// create a piece with deleted piece number and receiving node ID from the pointer
+			pieceToAdd[0] = &pb.RemotePiece{
+				PieceNum: pieceToRemove[0].PieceNum,
+				NodeId:   recNodeID,
+			}
+
+			_, err = satellite.Metainfo.Service.UpdatePieces(ctx, path, pointer, pieceToAdd, pieceToRemove)
+			require.NoError(t, err)
+
+			signer := signing.SignerFromFullIdentity(receivingIdentity)
+
+			signedNewPieceHash, err := signing.SignPieceHash(ctx, signer, newPieceHash)
+			require.NoError(t, err)
+
+			success := &pb.StorageNodeMessage{
+				Message: &pb.StorageNodeMessage_Succeeded{
+					Succeeded: &pb.TransferSucceeded{
+						OriginalPieceId:      m.TransferPiece.OriginalPieceId,
+						OriginalPieceHash:    originalPieceHash,
+						OriginalOrderLimit:   &orderLimit,
+						ReplacementPieceHash: signedNewPieceHash,
+					},
+				},
+			}
+			err = processClient.Send(success)
+			require.NoError(t, err)
+		default:
+			t.FailNow()
+		}
+
+		_, err = processClient.Recv()
+		require.Error(t, err)
+		require.True(t, errs2.IsRPC(err, rpcstatus.Internal))
```

Comment:
nit - should we check metainfo again at the end to make sure the recNodeID only shows up once?